crates/oxc_semantic/src/reference.rs

```diff
@@ -117,6 +117,6 @@ impl Reference {
     /// Returns `true` if this reference is used in a type context.
     #[inline]
     pub fn is_type(&self) -> bool {
-        self.flags.is_type() || self.flags.is_ts_type_query()
+        self.flags.is_type()
```

Comment:
this behavior is depended upon by `no-unused-vars`. If we want to change what `is_type` does, we need to re-add this API under a different name.

Reply:
maybe we should have an `is_used_as_type` method?

Reply:
The behavior has not changed

 `ReferenceFlags` that used to be `ReferenceFlags::Read` | `ReferenceFlags::TSTypeQuery` are now `ReferenceFlags::Type`.

Reply:
Nothing fails = happy Don